nixos/modules/services/torrent/deluge.nix

```diff
@@ -118,36 +118,86 @@ in {
             more informations.
           '';
         };
+
+        user = mkOption {
+          type = types.str;
+          default = "deluge";
+          description = ''
+            User account under which deluge runs.
+          '';
+        };
+
+        group = mkOption {
+          type = types.str;
+          default = "deluge";
+          description = ''
+            Group under which deluge runs.
+          '';
+        };
+
+        extraPackages = mkOption {
+          type = types.listOf types.package;
+          default = with pkgs; [ unzip gnutar xz p7zip bzip2 ];
```

Comment:
How it's implemented now (with a `defaults = [ ... ]`), whenever the user sets a value in their own config all the defaults get cleared. To prevent this you need to set these values in the `config` section of the module.

Reply:
I'm not quite sure I understand, but I've made a change, so let me know if that wasn't what you meant.

Reply:
Yeah that's what I meant.

The module system works with priorities. Only values with the highest priority get used (and merged together). By default the priority is 100, which is the case when users assign values in their configuration.nix without anything special. `mkDefault` lowers the priority to 1000, and `mkOptionDefault` lowers it to 1500. By assigning a default with `default = ` in the option, it uses the `mkOptionDefault` priority, which means these values get ignored as soon as anything with a higher priority exists, which is the case when users assign something in their config.